### PR TITLE
Merge main with feature/amqp

### PR DESCRIPTION
**Pull request by LarryOsterman**
# Pull Request Checklist

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/main/CONTRIBUTING.md#pull-requests).

- [ ] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [ ] Doxygen docs
- [ ] Unit tests
- [ ] No unwanted commits/changes
- [ ] Descriptive title/description
  - [ ] PR is single purpose
  - [ ] Related issue listed
- [ ] Comments in source
- [ ] No typos
- [ ] Update changelog
- [ ] Not work-in-progress
- [ ] External references or docs updated
- [ ] Self review of PR done
- [ ] Any breaking changes?

### CODE REVIEW

**Review by antkmsft — COMMENTED** [review]
I know that CI for this PR is failing, but if these changes are largely what this PR is going to contain, all look good to me so far.

File: eng/common/pipelines/templates/steps/run-pester-tests.yml [hunk 1]
```diff
@@ -30,26 +30,28 @@ steps:
...
+    displayName: Publish Code Coverage to Azure DevOps
     condition: succeededOrFailed()
     inputs:
-      codeCoverageTool: 'JaCoCo'
-      summaryFileLocation: '$(Build.SourcesDirectory)/${{ parameters.TargetDirectory }}/coverage.xml'
-      pathToSources: '$(Build.SourcesDirectory)/${{ parameters.TargetDirectory }}'
\ No newline at end of file
+      codeCoverageTool: JaCoCo
+      summaryFileLocation: $(Build.SourcesDirectory)/${{ parameters.TargetDirectory }}/coverage.xml
+      pathToSources: $(Build.SourcesDirectory)/${{ parameters.TargetDirectory }}
```
Comment (gearama) [inline eng/common/pipelines/templates/steps/run-pester-tests.yml:57]: new line

File: eng/common/scripts/get-codeowners.lib.ps1 [hunk 2]
```diff
@@ -0,0 +1,133 @@
...
+
+$codeownersJson = $commandOutput | ConvertFrom-Json
+  
+  if ($VsoVariable) {
+    $codeowners = $codeownersJson.Owners -join ","
+    Write-Host "##vso[task.setvariable variable=$VsoVariable;]$codeowners"
+  }
+
+  return ,@($codeownersJson.Owners)
+}
```
Comment (gearama) [inline eng/common/scripts/get-codeowners.lib.ps1:133]: new line

File: eng/common/scripts/get-codeowners.ps1 [hunk 3]
```diff
@@ -1,177 +1,18 @@
...
-  exit 0
-}
-else {
-  return Get-Codeowners -targetPath $TargetPath -targetDirectory $TargetDirectory -codeownersFileLocation $CodeOwnerFileLocation -includeNonUserAliases $IncludeNonUserAliases
-}
+return Get-Codeowners `
+  -TargetPath $TargetPath `
+  -TargetDirectory $TargetDirectory `
+  -CodeownersFileLocation $CodeownersFileLocation `
+  -IncludeNonUserAliases:$IncludeNonUserAliases
```
Comment (gearama) [inline eng/common/scripts/get-codeowners.ps1:18]: new line

**Review by gearama — APPROVED** [review]

File: eng/common/pipelines/templates/steps/run-pester-tests.yml (hunk 1 above)
Comment (LarryOsterman) [reply eng/common/pipelines/templates/steps/run-pester-tests.yml:57]: Not my code.

File: eng/common/scripts/get-codeowners.lib.ps1 (hunk 2 above)
Comment (LarryOsterman) [reply eng/common/scripts/get-codeowners.lib.ps1:133]: Not my code - this is in eng/common.

File: eng/common/scripts/get-codeowners.ps1 (hunk 3 above)
Comment (LarryOsterman) [reply eng/common/scripts/get-codeowners.ps1:18]: eng/common

**Pull request merged**